Pretty quick

chrismaree opened the pull request:
This PR adds `pretty-quick` to our linter. From my testing this decreases the time to lint before pushing by about half if not more as prettier only runs on files that have changed. I intentionally left the previous configs here such that CI will still lint the whole repo. 

Running `npm run lint` will still execute `prettier` and `eslint` over all js and .sol files as before. 

Future refinements of this could move the files we want to lint into the `.prettierrc` but I could not get this to work.

mrice32 commented on package.json:
```diff
@@ -5,12 +5,15 @@
-    "prettier_vanilla": "prettier --bracket-spacing 'core/**/*.js' 'common/**/*.js' '*.js' 'voter-dapp/src/**/*.js' 'sponsor-dapp-v2/src/**/*.js' 'core/contracts/**/*.sol' 'financial-templates-lib/*.js' 'liquidator/**/*.js' 'disputer/**/*.js'"
+    "prettier_check_quick": "npm run pretty_quick -- --check",
+    "prettier_vanilla": "prettier --bracket-spacing 'core/**/*.js' 'common/**/*.js' '*.js' 'voter-dapp/src/**/*.js' 'sponsor-dapp-v2/src/**/*.js' 'core/contracts/**/*.sol' 'financial-templates-lib/*.js' 'liquidator/**/*.js' 'disputer/**/*.js'",
+    "pretty_quick": "pretty-quick --bracket-spacing 'core/**/*.js' 'common/**/*.js' '*.js' 'voter-dapp/src/**/*.js' 'sponsor-dapp-v2/src/**/*.js' 'core/contracts/**/*.sol' 'financial-templates-lib/*.js' 'liquidator/**/*.js' 'disputer/**/*.js'"
```
This list is already kind of a nightmare to maintain. Do you think we could do something like this? Not sure if this actually works, but it might mean we don't have to maintain this list in two places.
```
"pretty_quick": "COMMAND=pretty-quick npm run prettier_vanilla",
"prettier_vanilla": "$COMMAND --bracket-spacing 'core/**/*.js' 'common/**/*.js' '*.js' 'voter-dapp/src/**/*.js' 'sponsor-dapp-v2/src/**/*.js' 'core/contracts/**/*.sol' 'financial-templates-lib/*.js' 'liquidator/**/*.js' 'disputer/**/*.js'"
```

chrismaree replied:
This is a better solution. I will give this ago. First prize for me is to put the files that it checks into the `.prettierrc` file.

mrice32 replied:
Is that possible?!?!

chrismaree commented:
> > only runs on files that have changed.
> 
> Changed as compared to what? The most recent commit? `master`? `origin/master`? Only unstaged changes?
> 
> It's not a huge deal if this isn't 100% correct because ci will still run the full check and catch if anything failed. I'd still like to better understand what it's comparing against and/or how it's configured.

It only runs on unstaged files locally. This is why I added it to the pre-push hook but did not modify the underlying `pretty_vanilla` check which runs within CI.

mrice32 commented:
> > > only runs on files that have changed.
> > 
> > 
> > Changed as compared to what? The most recent commit? `master`? `origin/master`? Only unstaged changes?
> > It's not a huge deal if this isn't 100% correct because ci will still run the full check and catch if anything failed. I'd still like to better understand what it's comparing against and/or how it's configured.
> 
> It only runs on unstaged files locally. This is why I added it to the pre-push hook but did not modify the underlying `pretty_vanilla` check which runs within CI.

Wait, unstaged *only*? Wouldn't that be fairly useless as a pre-push hook since it's not checking the files that you're pushing, only the ones that you haven't committed yet?

chrismaree commented:
> > > > only runs on files that have changed.
> > > 
> > > 
> > > Changed as compared to what? The most recent commit? `master`? `origin/master`? Only unstaged changes?
> > > It's not a huge deal if this isn't 100% correct because ci will still run the full check and catch if anything failed. I'd still like to better understand what it's comparing against and/or how it's configured.
> > 
> > 
> > It only runs on unstaged files locally. This is why I added it to the pre-push hook but did not modify the underlying `pretty_vanilla` check which runs within CI.
> 
> Wait, unstaged _only_? Wouldn't that not make sense as a pre-push hook since it's not checking the files that you're pushing, only the ones that you haven't committed yet?

Sorry, that was a typo 🤕 . I meant staged only. So it looks at the files you've check into your current commit and runs the linter against this.

mrice32 commented:
> > > > > only runs on files that have changed.
> > > > 
> > > > 
> > > > Changed as compared to what? The most recent commit? `master`? `origin/master`? Only unstaged changes?
> > > > It's not a huge deal if this isn't 100% correct because ci will still run the full check and catch if anything failed. I'd still like to better understand what it's comparing against and/or how it's configured.
> > > 
> > > 
> > > It only runs on unstaged files locally. This is why I added it to the pre-push hook but did not modify the underlying `pretty_vanilla` check which runs within CI.
> > 
> > 
> > Wait, unstaged _only_? Wouldn't that not make sense as a pre-push hook since it's not checking the files that you're pushing, only the ones that you haven't committed yet?
> 
> Sorry, that was a typo 🤕 . I meant staged only. So it looks at the files you've check into your current commit and runs the linter against this. [here](https://github.com/azz/pretty-quick#--staged-only-git) is the link in the offical docs.

Would we want `--branch` instead of staged: https://github.com/azz/pretty-quick#--branch? I thought staged files weren't committed yet?

chrismaree commented:
> > > > > > only runs on files that have changed.
> > > > > 
> > > > > 
> > > > > Changed as compared to what? The most recent commit? `master`? `origin/master`? Only unstaged changes?
> > > > > It's not a huge deal if this isn't 100% correct because ci will still run the full check and catch if anything failed. I'd still like to better understand what it's comparing against and/or how it's configured.
> > > > 
> > > > 
> > > > It only runs on unstaged files locally. This is why I added it to the pre-push hook but did not modify the underlying `pretty_vanilla` check which runs within CI.
> > > 
> > > 
> > > Wait, unstaged _only_? Wouldn't that not make sense as a pre-push hook since it's not checking the files that you're pushing, only the ones that you haven't committed yet?
> > 
> > 
> > Sorry, that was a typo 🤕 . I meant staged only. So it looks at the files you've check into your current commit and runs the linter against this. [here](https://github.com/azz/pretty-quick#--staged-only-git) is the link in the offical docs.
> 
> Would we want `--branch` instead of staged: [azz/pretty-quick#--branch](https://github.com/azz/pretty-quick#--branch)? I thought staged files weren't committed yet?

I think either is fine. From my testing staged ran on files that were committed but not pushed. I added the `--branch` flag as this is moreencompassingg.

mrice32 commented:
> > > > > > > only runs on files that have changed.
> > > > > > 
> > > > > > 
> > > > > > Changed as compared to what? The most recent commit? `master`? `origin/master`? Only unstaged changes?
> > > > > > It's not a huge deal if this isn't 100% correct because ci will still run the full check and catch if anything failed. I'd still like to better understand what it's comparing against and/or how it's configured.
> > > > > 
> > > > > 
> > > > > It only runs on unstaged files locally. This is why I added it to the pre-push hook but did not modify the underlying `pretty_vanilla` check which runs within CI.
> > > > 
> > > > 
> > > > Wait, unstaged _only_? Wouldn't that not make sense as a pre-push hook since it's not checking the files that you're pushing, only the ones that you haven't committed yet?
> > > 
> > > 
> > > Sorry, that was a typo 🤕 . I meant staged only. So it looks at the files you've check into your current commit and runs the linter against this. [here](https://github.com/azz/pretty-quick#--staged-only-git) is the link in the offical docs.
> > 
> > 
> > Would we want `--branch` instead of staged: [azz/pretty-quick#--branch](https://github.com/azz/pretty-quick#--branch)? I thought staged files weren't committed yet?
> 
> I think either is fine. From my testing staged ran on files that were committed but not pushed. I added the `--branch` flag as this is moreencompassingg.

I see -- if they do the same thing then either SGTM.
